src/en/07_ffi.md

```diff
@@ -623,12 +623,8 @@ pub struct Foo(*mut RawFoo);
 impl Foo {
     /// Create a Foo
     pub fn new() -> Option<Foo> {
-        let raw_ptr = unsafe { foo_create() };
-        if raw_ptr.is_null() {
-            None
-        } else {
-            Some(Foo(raw_ptr))
-        }
+        let raw_ptr = unsafe { foo_create().as_mut()? };
```

Comment:
The price to pay to being more Rusty is a potentially excessive terseness _w.r.t._ programmers perhaps not as acquainted with Rust as we'd expect them to be (they might be C/C++ programmers with little Rust experience that happened to have to interface with some Rust codebase).

So, to palliate this, especially in an example, let's try to be overly explicit regarding the role of `?`:

```suggestion
        // The `?` operator here ensures `None` is returned on `NULL`.
        let raw_ptr = unsafe { foo_create().as_mut()? };
```